code/espurna/ota_httpupdate.ino

```diff
@@ -0,0 +1,251 @@
+            break;
+        case HTTP_UPDATE_OK:
+            DEBUG_MSG_P(PSTR("[OTA] Done, restarting..."));
+            deferredReset(500, CUSTOM_RESET_OTA); // wait a bit more than usual
+            break;
+    }
+
+}
+
+#if OTA_CLIENT_HTTPUPDATE_2_3_0_COMPATIBLE
+void _otaClientFromHttp(const String& url) {
+    _otaClientRunUpdater(nullptr, url, "");
+}
+#else
+void _otaClientFromHttp(const String& url) {
+    auto client = std::make_unique<WiFiClient>();
+    _otaClientRunUpdater(client.get(), url, "");
+}
+#endif
+
+#if SECURE_CLIENT == SECURE_CLIENT_BEARSSL
+
+void _otaClientFromHttps(const String& url) {
+
+    int check = getSetting("otaScCheck", SECURE_CLIENT_CHECK_FINGERPRINT).toInt();
+    bool settime = (check == SECURE_CLIENT_CHECK_CA);
+
+    if (!ntpSynced() && settime) {
+        DEBUG_MSG_P(PSTR("[OTA] Time not synced!\n"));
+        return;
+    }
+
+    // unique_ptr self-destructs after exiting function scope
+    // create WiFiClient on heap to use less stack space
+    auto client = std::make_unique<BearSSL::WiFiClientSecure>();
+
+    if (check == SECURE_CLIENT_CHECK_NONE) {
+        DEBUG_MSG_P(PSTR("[OTA] !!! Connection will not be validated !!!\n"));
+        client->setInsecure();
+    }
+
+    if (check == SECURE_CLIENT_CHECK_FINGERPRINT) {
+        String fp_string = getSetting("otafp", OTA_FINGERPRINT);
+        if (!fp_string.length()) {
+            DEBUG_MSG_P(PSTR("[OTA] Requested fingerprint auth, but 'otafp' is not set\n"));
+            return;
+        }
+
+        uint8_t fp_bytes[20] = {0};
+        sslFingerPrintArray(fp_string.c_str(), fp_bytes);
+
+        client->setFingerprint(fp_bytes);
+    }
+
+    BearSSL::X509List *ca = nullptr;
+    if (check == SECURE_CLIENT_CHECK_CA) {
+        ca = new BearSSL::X509List(_ota_client_http_update_ca);
+        // because we do not support libc methods of getting time, force client to use ntpclientlib's current time
+        // XXX: local2utc method use is detrimental when DST happening. now() should be utc
+        client->setX509Time(ntpLocal2UTC(now()));
+        client->setTrustAnchors(ca);
+    }
+
+    // TODO: RX and TX buffer sizes must be equal?
+    const uint16_t requested_mfln = getSetting("otaScMFLN", OTA_SECURE_CLIENT_MFLN).toInt();
+    switch (requested_mfln) {
+        // default, do nothing
+        case 0:
+            break;
+        // match valid sizes only
+        case 512:
+        case 1024:
+        case 2048:
+        case 4096:
+        {
+            client->setBufferSizes(requested_mfln, requested_mfln);
+        }
+        default:
+            DEBUG_MSG_P(PSTR("[OTA] Warning: MFLN buffer size must be one of 512, 1024, 2048 or 4096\n"));
+    }
+
+    _otaClientRunUpdater(client.get(), url);
+
+}
+
+#endif // SECURE_CLIENT_BEARSSL
+
+
+#if SECURE_CLIENT == SECURE_CLIENT_AXTLS
+
+void _otaClientFromHttps(const String& url) {
+
+    const int check = getSetting("otaScCheck", SECURE_CLIENT_CHECK_FINGERPRINT).toInt();
```

Comment:
ref #1829 comment about the defaults... How about
general.h
```
#define SECURE_CLIENT_CHECK SECURE_CLIENT_CHECK_FINGERPRINT
...
#define OTA_SECURE_CLIENT_CHECK SECURE_CLIENT_CHECK // chained global default
```
so this uses the `SECURE_CLIENT_CHECK_FINGERPRINT` instead of trying to think about possible default values.

Reply:
Yes, I like this approach - will update my PR as well